compat/src/index.js

```diff
@@ -119,15 +130,27 @@ const flushSync = (callback, arg) => callback(arg);
 const StrictMode = Fragment;
 
 export function startTransition(cb) {
-	cb();
+	requestIdleCallbackPreact(() => {
+		cb();
+	});
 }
 
 export function useDeferredValue(val) {
 	return val;
 }
 
 export function useTransition() {
-	return [false, startTransition];
+	const [isTransitionPending, setIsTransitionPending] = useState(false);
+
+	const startTransitionFunction = (cb) => {
```

Comment:
The function identity will change with every render. Let's find a way to keep the reference stable.

Reply:
using use callback have any effect on it ?

> The function identity will change with every render. Let's find a way to keep the reference stable.